matching-core/src/main/java/com/graphhopper/matching/MatchEdge.java

```diff
@@ -0,0 +1,38 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for 
+ *  additional information regarding copyright ownership.
+ * 
+ *  GraphHopper GmbH licenses this file to you under the Apache License, 
+ *  Version 2.0 (the "License"); you may not use this file except in 
+ *  compliance with the License. You may obtain a copy of the License at
+ * 
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.matching;
+
+import com.graphhopper.util.EdgeIteratorState;
+
+/**
+ *
+ * @author kodonnell
+ */
+public class MatchEdge {
+
+    public final EdgeIteratorState edge;
+    public final long fromTime;
+    public final long toTime;
+
+    public MatchEdge(EdgeIteratorState edge, long fromTime, long toTime) {
+        assert edge != null : "edge should not be null";
```

Comment:
If a precondition of a public method is violated, an exception should rather be thrown (NullPointerException in this case).